### ISSUE TITLE
Use enums for setting representation and display_preference

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#1024]
The [`representation` property in Appearance](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/Appearance.h#L60) and [`display_preference` property in ContactGeometry](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/ContactGeometry.h#L59) are currently ints, but I think it would be better to use enums so that code like
```cpp
myContactSphere->setDisplayPreference(4); //requires a comment
```
becomes
```cpp
myContactSphere->setDisplayPreference(ContactGeometry::shaded); //self-documenting
```
or use Simbody's SimTK::DecorativeGeometry::Representation enum:
```cpp
myContactSphere->setDisplayPreference(SimTK::DecorativeGeometry::DrawSurface);
```

**Comment by tkuchida** [comment]
Hm, the sphere is always drawn in wireframe anyway (see [here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/ContactSphere.cpp#L98)). Seems like an easy fix to select the [SimTK::DecorativeGeometry::Representation](https://simtk.org/api_docs/simbody/latest/classSimTK_1_1DecorativeGeometry.html#a142c12528e572a362c2b0535686cd50a) that corresponds to the display_preference property.

**Comment by aymanhab** [comment]
@tkuchida Thanks for trying this. Ideally we define the enum and map it to int so you use the descriptive enum in code and when we serialize we don't deal with unnecessary typos/case-sensitivity/padding issues related to strings. Indeed we can use the preference in display_preference if ContactSphere has one. As of now ContactGeometry has its own display_preference while other Geometry have Appearance. These are in the process of unification ;) so they can be leveraged, edited/serialized using the same code.

**Comment by jenhicks** [comment]
@aymanhab Can you check that this has been fixed?

**Comment by aymanhab** [comment]
These are all fixed by PR #1122 still in the pipe.
